File: packages/contract-npm/src/events.ts
export enum TransportEventType {
  HTTP_REQUEST = "http.request",
  HTTP_DISCONNECT = "http.disconnect",
  HTTP_RESPONSE_START = "http.response.start",
  HTTP_RESPONSE_BODY = "http.response.body",
  HTTP_RESPONSE_PATHSEND = "http.response.pathsend",
  WEBSOCKET_CONNECT = "websocket.connect",
  WEBSOCKET_RECEIVE = "websocket.receive",
  WEBSOCKET_DISCONNECT = "websocket.disconnect",
  WEBSOCKET_ACCEPT = "websocket.accept",
  WEBSOCKET_SEND = "websocket.send",
  WEBSOCKET_CLOSE = "websocket.close",
  WEBTRANSPORT_CONNECT = "webtransport.connect",
  WEBTRANSPORT_ACCEPT = "webtransport.accept",
  WEBTRANSPORT_STREAM_RECEIVE = "webtransport.stream.receive",
  WEBTRANSPORT_STREAM_SEND = "webtransport.stream.send",
  WEBTRANSPORT_DATAGRAM_RECEIVE = "webtransport.datagram.receive",
  WEBTRANSPORT_DATAGRAM_SEND = "webtransport.datagram.send",
  WEBTRANSPORT_DISCONNECT = "webtransport.disconnect",
  WEBTRANSPORT_CLOSE = "webtransport.close",
  LIFESPAN_STARTUP = "lifespan.startup",
  LIFESPAN_STARTUP_COMPLETE = "lifespan.startup.complete",
  LIFESPAN_STARTUP_FAILED = "lifespan.startup.failed",
  LIFESPAN_SHUTDOWN = "lifespan.shutdown",
  LIFESPAN_SHUTDOWN_COMPLETE = "lifespan.shutdown.complete",
  LIFESPAN_SHUTDOWN_FAILED = "lifespan.shutdown.failed",
  TRANSPORT_EMIT_COMPLETE = "transport.emit.complete",
}
export interface ContractEvent {
  type: TransportEventType;
  message?: string | null;
  payload?: Record<string, unknown>;
}

File: packages/contract-rs/src/events.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Copy, PartialEq, Eq, Serialize, Deserialize)]
pub enum TransportEventType {
    #[serde(rename = "http.request")]
    HttpRequest,
    #[serde(rename = "http.disconnect")]
    HttpDisconnect,
    #[serde(rename = "http.response.start")]
    HttpResponseStart,
    #[serde(rename = "http.response.body")]
    HttpResponseBody,
    #[serde(rename = "http.response.pathsend")]
    HttpResponsePathsend,
    #[serde(rename = "websocket.connect")]
    WebsocketConnect,
    #[serde(rename = "websocket.receive")]
    WebsocketReceive,
    #[serde(rename = "websocket.disconnect")]
    WebsocketDisconnect,
    #[serde(rename = "websocket.accept")]
    WebsocketAccept,
    #[serde(rename = "websocket.send")]
    WebsocketSend,
    #[serde(rename = "websocket.close")]
    WebsocketClose,
    #[serde(rename = "webtransport.connect")]
    WebtransportConnect,
    #[serde(rename = "webtransport.accept")]
    WebtransportAccept,
    #[serde(rename = "webtransport.stream.receive")]
    WebtransportStreamReceive,
    #[serde(rename = "webtransport.stream.send")]
    WebtransportStreamSend,
    #[serde(rename = "webtransport.stream.close")]
    WebtransportStreamClose,
    #[serde(rename = "webtransport.stream.reset")]
    WebtransportStreamReset,
    #[serde(rename = "webtransport.stream.stop_sending")]
    WebtransportStreamStopSending,
    #[serde(rename = "webtransport.datagram.receive")]
    WebtransportDatagramReceive,
    #[serde(rename = "webtransport.datagram.send")]
    WebtransportDatagramSend,
    #[serde(rename = "webtransport.disconnect")]
    WebtransportDisconnect,
    #[serde(rename = "webtransport.close")]
    WebtransportClose,
    #[serde(rename = "lifespan.startup")]
    LifespanStartup,
    #[serde(rename = "lifespan.startup.complete")]
    LifespanStartupComplete,
    #[serde(rename = "lifespan.startup.failed")]
    LifespanStartupFailed,
    #[serde(rename = "lifespan.shutdown")]
    LifespanShutdown,
    #[serde(rename = "lifespan.shutdown.complete")]
    LifespanShutdownComplete,
    #[serde(rename = "lifespan.shutdown.failed")]
    LifespanShutdownFailed,
    #[serde(rename = "transport.emit.complete")]
    TransportEmitComplete,
    #[serde(rename = "transport.emit.failed")]
    TransportEmitFailed,
}

impl Default for TransportEventType {
    fn default() -> Self {
        Self::HttpRequest
    }
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ContractEvent {
    pub r#type: TransportEventType,
    pub message: Option<String>,
    pub payload: serde_json::Map<String, serde_json::Value>,
}
